src/classes/Pantry.js

```diff
@@ -0,0 +1,23 @@
+class Pantry {
+  constructor(ingredients) {
+    this.ingredients = ingredients
+  }
+
+  checkPantry(recipe) {
+    let hasAllIngredients = true;
+
+    recipe.ingredients.forEach(recipeIng => {
+      const ingInPantry = this.ingredients.find(pantryIng => {
+        return recipeIng.id === pantryIng.ingredient;
+      })
+
+      if ((ingInPantry && (ingInPantry.amount < recipeIng.quantity.amount)) || !ingInPantry) {
```

Comment:
The magic of parens. We had the logic here but had to keep re-working till we realized we were missing a set of parens. Yippee!